src/cliffordsynthesis/Tableau.cpp

```diff
@@ -322,4 +310,47 @@ Tableau::Tableau(const qc::QuantumComputation& qc, const std::size_t begin,
     }
   }
 }
+void Tableau::fromString(const std::string& stabilizers,
+                         const std::string& destabilizers) {
+  loadStabilizerDestabilizerString(destabilizers);
+  loadStabilizerDestabilizerString(stabilizers);
+}
+void Tableau::loadStabilizerDestabilizerString(const std::string& string) {
+  std::stringstream ss(string);
+  std::string       line;
+  std::getline(ss, line);
+  if (line.empty()) {
+    return;
+  }
+  const auto  rStabilizer = std::regex("([\\+-]?)([IYZX]+)");
+  std::smatch m;
+  // string is a list of stabilizers
+  auto iter = line.cbegin();
+  while (std::regex_search(iter, line.cend(), m, rStabilizer)) {
+    std::string s = m.str(0U);
+    RowType     row;
+
+    for (const auto c : s) {
+      if (c == 'I' || c == 'Z') {
+        row.push_back(0);
+      } else if (c == 'X' || c == 'Y') {
+        row.push_back(1);
+      }
+    }
+    for (const auto c : s) {
+      if (c == 'I' || c == 'X') {
+        row.push_back(0);
+      } else if (c == 'Y' || c == 'Z') {
+        row.push_back(1);
+      }
+    }
+    if (s[0U] == '-') {
+      row.push_back(1);
+    } else {
+      row.push_back(0);
+    }
+    tableau.push_back(row);
+    iter = m[0].second;
+  }
+}
```

Comment:
Can we do some more validation here?
Stabilizers should be a list.
"I" also should have an alternative representation as "_".
Otherwise an error should be thrown.